pkgs/development/libraries/libidn2/default.nix

```diff
@@ -20,9 +22,19 @@ stdenv.mkDerivation rec {
 
   patches = optional stdenv.isDarwin ./fix-error-darwin.patch;
 
+  # hack found here: https://github.com/sethrj/spack/commit/93393fe2f510a8ed9ace65534ee80e5433e19943
+  # works around the fact that autoreconf'ing libidn2 causes it to insist on gtk-doc being present
+  GTKDOCIZE = "echo";
+
   # The above patch causes the documentation to be regenerated, so the
   # documentation tools are required.
-  nativeBuildInputs = optionals stdenv.isDarwin [ help2man texinfo ];
+  nativeBuildInputs = [
+    # libidn2's Makefile forces an autoreconf if it notices changes to the ./configure script
+    autoreconfHook buildPackages.pkg-config
+  ] ++  optionals stdenv.isDarwin [
+    help2man texinfo
```

Comment:
Because GTKDOCIZE is now always overridden, are these tools still needed for darwin?

Reply:
Good question; I don't really know.  I don't have Darwin hardware to test so I wanted to err on the side of not breaking anything.

Reply:
That's fair.